### PR TITLE
[migration] Fix multiport Services broken by Helm

### PR DESCRIPTION
**Pull request by tarabrind**
## Description
Fixing the broken Services with multiple ports. Helm didn't handle Service resource modification well while deploying https://github.com/deckhouse/deckhouse/pull/8955.

## Why do we need it, and what problem does it solve?
In some clusters, Services became broken. In some clusters it was kube-dns.

## What is the expected result?
All Services are corresponded to their templates.

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
---
section: kube-dns
type: fix
summary: Fixed the d8-kube-dns and d8-kube-dns-redirect Services with multiple ports broken by Helm (pr #8955).
---
section: delivery
type: fix
summary: Fixed the argocd-repo-server and argocd-server Services with multiple ports broken by Helm (pr #8955).
---
section: deckhouse
type: fix
summary: Fixed the deckhouse-leader and deckhouse Services with multiple ports broken by Helm (pr #8955).
---
section: istio
type: fix
summary: Fixed the kiali Service with multiple ports broken by Helm (pr #8955).
---
section: prometheus
type: fix
summary: Fixed the memcached Service with multiple ports broken by Helm (pr #8955).
```

### CODE REVIEW

**Review by Beastlex — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @tarabrind. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/002-deckhouse/templates/service.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ kind: Service
 metadata:
   name: deckhouse
   namespace: d8-system
-  {{- include "helm_lib_module_labels" (list . (dict "app" "deckhouse")) | nindent 2 }}
+  {{- include "helm_lib_module_labels" (list . (dict "app" "deckhouse" "service-helm-fix" "true")) | nindent 2 }}
```
Comment (apolovov) [inline modules/002-deckhouse/templates/service.yaml:7]:
```suggestion
  {{- include "helm_lib_module_labels" (list . (dict "app" "deckhouse" "migration.deckhouse.io/fix-services-broken-by-helm" "done")) | nindent 2 }}
```

File: modules/110-istio/hooks/migration_service_with_many_ports.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnBeforeHelm: &go_hook.OrderedConfig{Order: 20},
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:       "service_helm_fix",
+			ApiVersion: "v1",
+			Kind:       "Service",
+			FilterFunc: applyServiceFilterHelmFix,
+			NameSelector: &types.NameSelector{
```
Comment (apolovov) [inline modules/110-istio/hooks/migration_service_with_many_ports.go:45]:
```suggestion
                         LabelSelector: &v1.LabelSelector{
                                MatchExpressions: []v1.LabelSelectorRequirement{
                                        {
                                                Key:      "migration.deckhouse.io/fix-services-broken-by-helm",
                                                Operator: v1.LabelSelectorOpNotIn,
                                                Values:   []string{"done"},
                                        },
                                },
                        },
			NameSelector: &types.NameSelector{
```

**Review by apolovov — CHANGES_REQUESTED** [review]
Discussed.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @tarabrind. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by apolovov — APPROVED** [review]

**Pull request merged**